### PR TITLE
AB#4135 - add userId as session variable after login

### PR DESCRIPTION
**Pull request by leiye29**
### Description

add userId as session variable after login

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/$lang/_protected/stub-sin-editor.tsx [hunk 1]
```diff
@@ -93,6 +94,10 @@ export async function action({ context: { session }, params, request }: ActionFu
   }
   session.set('idToken', idToken);
 
+  const subscriptionService = getSubscriptionService();
+  const userInformation = await subscriptionService.getUserByRaoidcUserId(session.get('userInfoToken').sub);
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/routes/$lang/_protected/stub-sin-editor.tsx:98]: const userInfoToken: UserinfoToken = session.get('userInfoToken');
  if (!session.has('userInfoToken')) {
    session.set('userInfoToken', userinfoTokenPayload);
  } else {
    userInfoToken.sin = sinToMock;
    userInfoToken.sub = userStubUUID;
    session.set('userInfoToken', userInfoToken);
  }
  session.set('idToken', idToken);

The userInfoToken has been grabbed from the session could be null, i need to grab it again after session reset `userInfoToken`

[the remark this answers is not in the document]
Comment (nicholas-ly) [reply frontend/app/routes/$lang/_protected/stub-sin-editor.tsx:98]: Understood - thanks. Resolve.

File: frontend/app/services/subscription-service.server.ts [hunk 2]
```diff
@@ -106,11 +141,11 @@ function createSubscriptionService() {
       throw new Error(`Failed to find data. Status: ${response.status}, Status Text: ${response.statusText}`);
     }
 
-    const users = await response.json();
-    if (users._embedded.users.length === 0) {
-      return null;
+    const user = await response.json();
+    if (!user) {
+      throw new Error(`Failed to find the user: ${userId}.`);
     }
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/services/subscription-service.server.ts]: removed

File: frontend/app/services/subscription-service.server.ts [hunk 3]
```diff
@@ -75,6 +75,41 @@ export const getSubscriptionService = moize(createSubscriptionService, { onCache
 function createSubscriptionService() {
   const { CDCP_API_BASE_URI } = getEnv();
 
+  async function getUserByRaoidcUserId(userId: string) {
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/services/subscription-service.server.ts:79]: added

**Review by nicholas-ly — APPROVED** [review]

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**